kong/conf_loader.lua

```diff
@@ -597,6 +598,34 @@ local function check_and_infer(conf, opts)
   -- custom validations
   ---------------------
 
+  conf.host_ports = {}
+  if conf.port_map then
+    local MIN_PORT = 1
+    local MAX_PORT = 65535
+
+    for _, ports in ipairs(conf.port_map) do
+      local colpos = string.find(ports, ":", nil, true)
+      if not colpos then
+        errors[#errors + 1] = "invalid port mapping (`port_map`): " .. ports
+
+      else
+        local host_port_str = string.sub(ports, 1, colpos - 1)
+        local host_port_num = tonumber(host_port_str, 10)
+        local kong_port_str = string.sub(ports, colpos + 1)
+        local kong_port_num = tonumber(kong_port_str, 10)
+
+        if  (host_port_num and host_port_num >= MIN_PORT and host_port_num <= MAX_PORT)
+        and (kong_port_num and kong_port_num >= MIN_PORT and kong_port_num <= MAX_PORT)
+        then
```

Comment:
Can we add a check to ensure that kong is configured to listen on kong_port_num?
I'm not sure how much scope creep happens because of this. I'll leave that up to you, Aapo.

Reply:
@hbagdi I looked at it, and it is a bit complicated (listeners are parsed later). The good side is that if the port is not the one that Kong is not listening, it will not do anything. It is just excess configuration. I will create a card about this.